core/src/test/kotlin/org/jetbrains/jewel/PainterHintTest.kt

```diff
@@ -49,74 +55,192 @@ class PainterHintTest : BasicJewelUiTest() {
         return result
     }
 
+    private val documentBuilderFactory = DocumentBuilderFactory.newDefaultInstance()
+        .apply { setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true) }
+
+    private fun String.applyPaletteHints(vararg hints: PainterHint): String {
+        val doc = documentBuilderFactory.newDocumentBuilder().parse(this.toByteArray().inputStream())
+
+        hints.forEach {
+            when (it) {
+                is PainterSvgPatchHint -> it.patch(doc.documentElement)
+                else -> return@forEach
+            }
+        }
```

Comment:
Do you want to interrupt processing when it's not an SvgPatchHint? Or were you going for something like this?

```suggestion
        hints.filterIsInstance(PainterSvgPatchHint::class)
            .onEach { it.patch(doc.documentElement) }
```